src/main.rs

```diff
@@ -563,18 +572,33 @@ fn run_cli(cli: Cli) -> Result<()> {
                 }
             };
 
-            if edit {
+            if !no_alias {
+                match Alias::from_str(&id) {
+                    Ok(alias) => {
+                        info!("Creating alias '{alias}' for '{canonical}'");
+                        row.apply(state::add_alias(&alias))?;
```

Comment:
The `add_alias` operation is fallible (if the alias exists, it returns `false`).

1. Should the return type of `add_alias` be unpacked into an actual enum, for clarity?
2. We should definitely do something if alias creation fails.

Reply:
I overlooked this. Will push a commit to fix this now.

I tried tagging `add_alias` with `#[must_use]` but it doesn't work because the return type is `Result`, which is used (and in fact, all `Result`s are must-use). I don't know if we can require the `Ok` variant to be used.

I think it's fine to leave it as `bool`.

Reply:
If you create a custom return type and annotate it with `#[must_use]`, what happens?

Reply:
Does the `#[must_use]` "reach downwards"? Or is it just forgotten immediately when unwrapping the result?

Reply:
I think that's the problem. Unwrapping or propogating seems enough to satify the compiler. Creating a custom return type for `add_refs_impl` and annotating it with `#[must_use]` works though; it creates 5 warnings at other lines.